libsql-server/src/connection/write_proxy.rs

```diff
@@ -332,7 +338,23 @@ where
                         continue;
                     }
 
-                    if !response_cb(resp.response.ok_or(Error::PrimaryStreamMisuse)?)? {
+                    let should_continue = tokio::task::spawn_blocking({
+                        let cb_context = cb_context.clone();
+                        move || -> Result<bool> {
+                            let mut ctx = cb_context.lock();
+                            let (ref mut response_cb, ref mut arg) = *ctx;
+                            if !response_cb(resp.response.ok_or(Error::PrimaryStreamMisuse)?, arg)?
+                            {
+                                Ok(false)
+                            } else {
+                                Ok(true)
+                            }
```

Comment:
that took quite some brainpower for me to parse, with all the negatives (: Isn't all this reducible to just:
```rust
response_cb(resp.response.ok_or(Error::PrimaryStreamMisuse)?, arg)
```
, after you drop the negations and collapse the `Ok(something?)` pattern?